lib/src/parallel_prefix_operations.dart

```diff
@@ -0,0 +1,255 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// parallel-prefix_operations.dart
+// Implementation of operators using various parallel-prefix trees.
+//
+// 2023 Sep 29
+// Author: Desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+//   Borrowed from https://github.com/stevenmburns/rohd_sklansky.git
+
+import 'dart:math';
+
+import 'package:collection/collection.dart';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// This computes the power of 2 less than x
+int largestPow2LessThan(int x) => pow(2, log2Ceil(x) - 1).toInt();
+
+/// ParallePrefix is the core parallel prefix tree structure node
+/// The output is a List of multi-bit Logic vectors (typicall 2-bit) that
+/// represent things like carry-save or generate-propagate signaling in adder
+/// networks.  Each node in a parallel prefix tree transforms a row of inputs
+/// to an equal length row of outputs of these multi-bit Logic values.
+class ParallelPrefix extends Module {
+  final List<Logic> _oseq = [];
+
+  /// Output sequence value
+  List<Logic> get val => _oseq;
```

Comment:
I think wrap `_oseq` so that it cant be modified by users: `UnmodifiableListView(_oSeq)`.